qiskit/pulse/instruction_schedule_map.py

```diff
@@ -267,3 +281,12 @@ def _to_tuple(values: Union[int, Iterable[int]]) -> Tuple[int, ...]:
         return tuple(values)
     except TypeError:
         return (values,)
+
+
+def _get_instruction_string(inst: Union[str, Gate]):
+    if isinstance(inst, str):
+        return inst
+    else:
+        assert(isinstance(inst, Gate)), "Instruction must be Gate obj or str"
+        assert(inst.name), "Gate has no name"
+        return inst.name
```

Comment:
This could be something like
```suggestion
        try:
            return inst.name
        except AttributeError:
            raise PulseError('Input "inst" has no attribute "name". This should be a circuit "Instruction".')
```